typing/typecore.ml

```diff
@@ -2811,8 +2856,28 @@ and type_pat_aux
       let alloc_mode =
         cross_left !!penv expected_ty alloc_mode.mode
       in
+      let mode, kind =
+        match mutable_flag with
+        | Immutable -> alloc_mode, Val_reg
+        | Mutable ->
+            let m0 = Value.Comonadic.newvar () in
+            let mode = mutvar_mode ~loc ~env:!!penv m0 alloc_mode in
+            let kind =
+              Val_mut
+              (* CR-someday let_mutable: move the sort calculation elsewhere *)
+              (m0,
+              match
```

Comment:
I think having a large expression like this as an element of a pair makes this a bit hard to read, and suggest pulling it out into a let binding.